neo/io/neomatlabio.py

```diff
@@ -351,69 +413,139 @@ def create_ob_from_struct(self, struct, classname):
 
             if attrname in dict_attributes:
                 attrtype = dict_attributes[attrname][0]
                 if attrtype == datetime:
-                    m = r'(\d+)-(\d+)-(\d+) (\d+):(\d+):(\d+).(\d+)'
+                    m = r"(\d+)-(\d+)-(\d+) (\d+):(\d+):(\d+).(\d+)"
                     r = re.findall(m, str(item))
                     if len(r) == 1:
                         item = datetime(*[int(e) for e in r[0]])
                     else:
                         item = None
                 elif attrtype == np.ndarray:
                     dt = dict_attributes[attrname][2]
-                    item = item.astype(dt)
+                    try:
+                        item = item.astype(dt)
+                    except AttributeError:
+                        # it seems arrays of length 1 are stored as scalars
+                        item = np.array([item], dtype=dt)
                 elif attrtype == pq.Quantity:
                     ndim = dict_attributes[attrname][1]
-                    units = str(getattr(struct, attrname + '_units'))
+                    units = str(getattr(struct, attrname + "_units"))
                     if ndim == 0:
                         item = pq.Quantity(item, units)
                     else:
                         item = pq.Quantity(item, units)
                 elif attrtype == dict:
-                    # FIXME: works but doesn't convert nested struct to dict
-                    item = {fn: getattr(item, fn) for fn in item._fieldnames}
+                    new_item = {}
+                    for fn in item._fieldnames:
+                        value = getattr(item, fn)
+                        if value == PY_NONE:
+                            value = None
+                        new_item[fn] = value
+                    item = new_item
                 else:
                     item = attrtype(item)
 
             setattr(ob, attrname, item)
 
+        neo_id = getattr(struct, "neo_id", None)
+        if neo_id:
+            setattr(ob, "_id", neo_id)
         return ob
+
+    def _resolve_references(self, bl):
+        if bl.groups:
+            obj_lookup = {}
+            for ob in bl.children_recur:
+                if hasattr(ob, "_id"):
+                    obj_lookup[ob._id] = ob
+            for grp in bl.groups:
+                for container_name in grp._child_containers:
+                    container = getattr(grp, container_name)
+                    for i, item in enumerate(container):
+                        if isinstance(item, _Ref):
+                            assert isinstance(item.identifier, (int, np.integer))
+                            # A reference to an object that already exists
+                            container[i] = obj_lookup[item.identifier]
+                        else:
+                            # ChannelView and RegionOfInterest
+                            assert item.is_view
```

Comment:
I think this is a different case, because it's purely internal, should never be triggered, and will only be triggered by someone modifying the code, in which case they can use a debugger

Reply:
Cool. Sounds good.